smart-contracts/wasm-chain-integration/src/v1/mod.rs

```diff
@@ -303,6 +335,8 @@ mod host {
     const QUERY_ACCOUNT_BALANCE_TAG: u32 = 2;
     const QUERY_CONTRACT_BALANCE_TAG: u32 = 3;
     const QUERY_EXCHANGE_RATE_TAG: u32 = 4;
+    const CHECK_ACCOUNT_SIGNATURE_TAG: u32 = 5;
+    const QUERY_ACCOUNT_KEYS_TAG: u32 = 6;
```

Comment:
Why are these 5 and 6, when they are 6 and 7 in `to_bytes` above? Seems like a constant for `upgrade` is missing.

Reply:
Right, the reason for that is that `upgrade` is not handled by the `invoke`, it is instead a separate host function so that it is visible from the contract code that it is being called.

But internally we handle it via the `invoke` handler. 

The other tags in `to_bytes` are purely an internal thing in how we pass data through FFI.  Whereas the tags here are part of the smart contract API.

Reply:
There are two places where we have to serialize the invoke call, Wasm to rust and rust to Haskell, since upgrade is a separate host function there is no tag for it in Wasm to rust, but there still is in the rust to Haskell part.

Reply:
Ah ok, thanks for the explanation. 👍